### PR TITLE
Limit cluster role access to certificates.k8s.io via config param

### PR DESCRIPTION
**Pull request by kfaseela**
The CSR related clusterroles are needed only for https://istio.io/latest/docs/tasks/security/cert-management/custom-ca-k8s/ which is an experimental feature.
Moreover as per : https://github.com/kubernetes-sigs/multi-tenancy/issues/1002 legacy-unknown resourceName was recommended to be removed.

So would it make sense to move this behind a config parameter?
Internally we have different analysis going on, to limit the cluster wide permissions as much as possible, and hence this attempt.

### CODE REVIEW

**Review by jacob-delgado — APPROVED** [review]

File: manifests/charts/default/values.yaml [hunk 1]
```diff
@@ -13,6 +13,9 @@ base:
   # Users should consider enabling this if they want full gateway-api validation but don't have other validation servers.
   validateGateway: false
 
+  # If disabled, permissions required to support custom k8s CAs will not be enabled.
+  enableCertificateSigningRequestControllers: true
```
Comment (howardjohn) [inline manifests/charts/default/values.yaml:17]: I think we already have existing APIs around this, do we need a new setting? I think EXTERNAL_CA enables it and PILOT_CERT_PROVIDER sets the signer name.

Also we have K8S_SIGNER, not sure why we have 2

cc @shankgan @irisdingbj

Comment (irisdingbj) [reply manifests/charts/default/values.yaml:17]: yes, we used EXTERNAL_CA  and PILOT_CERT_PROVIDER for external CA integration.  Not sure K8S_SIGNER though. @shankgan might share more lights here?

Comment (kfaseela) [reply manifests/charts/default/values.yaml]: @irisdingbj @howardjohn : i removed the additional parameter, and reused EXTERNAL_CA and defaultCertProvider, could you please have a look now?

File: manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml [hunk 2]
```diff
@@ -71,6 +71,7 @@ rules:
     verbs: ["create", "get", "list", "watch", "update"]
 
   # Istiod and bootstrap.
+{{- if or .Values.pilot.env.EXTERNAL_CA (ne .Values.global.pilotCertProvider "istiod") }}
```
Comment (howardjohn) [inline manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml:74]: can we ignore all of "istiod", "custom", "none"?

Comment (kfaseela) [reply manifests/charts/istio-control/istio-discovery/templates/clusterrole.yaml]: Done

**Review by howardjohn — APPROVED** [review]

File: tests/integration/security/external_ca/main_test.go [hunk 3]
```diff
@@ -92,8 +95,6 @@ components:
       env:
       - name: CERT_SIGNER_DOMAIN
         value: clusterissuers.istio.io
-      - name: EXTERNAL_CA
```
Comment (irisdingbj) [inline tests/integration/security/external_ca/main_test.go:95]: Can we keep this?  since we have users depend on this setting already

Comment (kfaseela) [reply tests/integration/security/external_ca/main_test.go:95]: I have updated the docs to align with this: https://github.com/istio/istio.io/pull/12908

**Pull request merged**